actionpack/lib/action_controller/metal/rendering.rb

```diff
@@ -4,6 +4,17 @@ module Rendering
 
     RENDER_FORMATS_IN_PRIORITY = [:body, :text, :plain, :html]
 
+    module ClassMethods
+      # Documentation at ActionController::Renderer#render
+      delegate :render, to: :renderer
+
+      # Returns a renderer class (inherited from ActionController::Renderer)
+      # for the controller.
+      def renderer
+        @renderer ||= Renderer.for(self)
```

Comment:
@egilburg each controller represents a different class, ivars won't be shared between them. `@renderer` will be set and accessible only inside context of the current controller class.